randovania/games/am2r/generator/pool_creator.py

```diff
@@ -35,42 +52,8 @@ def artifact_pool(game: GameDescription, configuration: AM2RConfiguration, rng:
     if config.required_artifacts > max_artifacts:
         raise InvalidConfiguration("More Metroid DNA than allowed!")
 
-    new_assignment: dict[PickupIndex, PickupEntry] = {}
-
     keys: list[PickupEntry] = [create_am2r_artifact(i, game.resource_database) for i in range(46)]
-
     keys_to_shuffle = keys[:config.required_artifacts]
     starting_keys = keys[config.required_artifacts:]
 
-    locations: list[PickupNode] = []
-    for node in game.region_list.all_nodes:
-        if isinstance(node, PickupNode):
-            # Metroid pickups
-            name = node.extra["object_name"]
-            if config.prefer_metroids and name.startswith("oItemDNA_"):
-                locations.append(node)
-            # Pickups guarded by bosses
-            elif config.prefer_bosses and name in _boss_items:
-                locations.append(node)
-
-    item_pool = keys_to_shuffle
-
-    if rng is not None:
-        rng.shuffle(locations)
-        new_assignment = {
-            location.pickup_index: key
-            for location, key in zip(locations, keys_to_shuffle)
-        }
-        item_pool = [key for key in keys_to_shuffle if key not in new_assignment.values()]
-
-    return PoolResults(item_pool, new_assignment, starting_keys)
-
-
-_boss_items = [
-    "oItemM_111",
-    "oItemJumpBall",
-    "oItemSpaceJump",
-    "oItemPBeam",
-    "oItemIBeam",
-    "oItemETank_50"
-]
+    return PoolResults(keys_to_shuffle, {}, starting_keys)
```

Comment:
For AM2R I need to shuffle the `keys` list, [in order to not have very predictable hints in-game](https://github.com/randovania/randovania/pull/4721/files#diff-beb11d84256c963bed4aa3521fc664616f21ef0d504cf3f04e42af55e07db312). What's the best place to do that now in / should that be done differently in any way?
My current naive line of thought is to just keep this here as-is for the gui (and whatever else needs it) to work, and then in `assign_pool_results` reevaluate again which dna numbers are starting.

Reply:
My counter question is why your hints are predictable is that list is not sorted?

Reply:
The way I currently have hints implemented in AM2R, is that each area has a hint location, that gives out hints for DNA which would all normally be in that area. So the hint location in Area 1 will always give out hints for DNA 2, 3, 4, 5. If this list is now sorted, and you have a preset with 5 shuffled dna, that means that dna 1, 2, 3, 4, 5 are shuffled. Which means in all seeds generated from that preset, the only hint locations you ever need to visit is the one in area 1, and the one that spoils the location for dna 1. All other hint locations would just give you the "you already start with this item" hint, thus becoming useless.

If the list is shuffled, then the keys are shuffled too, so on every seed generated from that preset, you'd have to visit different hint locations to get all needed hints, and not just always the same locations.

Reply:
Addressed by shuffling the keys before assigning them.